mypy/newsemanal/semanal.py

```diff
@@ -348,6 +348,20 @@ def prepare_file(self, file_node: MypyFile) -> None:
                                                               self.modules['builtins'])
         if file_node.fullname() == 'builtins':
             self.prepare_builtins_namespace(file_node)
+        if file_node.fullname() == 'typing':
+            self.prepare_typing_namespace(file_node)
+
+    def prepare_typing_namespace(self, file_node: MypyFile) -> None:
+        """Remove dummy alias definitions.
```

Comment:
This would be more concrete if with an example, such as `List = TypeAlias(object).`